R/combinedscatterplot.R

```diff
@@ -0,0 +1,827 @@
+
+    output <- getColors(colors, scatter.colors, n, not.na,
+                        scatter.colors.as.categorical, num.tables, legend.show,
+                        groups)
+    colors <- output$colors
+    scatter.colors <- output$scatter.colors
+    groups <- output$groups
+    legend.show <- output$legend.show
+    not.na <- output$not.na
+
+    if (is.na(data.label.font.autocolor)) {
+        data.label.font.autocolor <- length(unique(groups[not.na])) > 1
+    }
+
+    if (trend.lines) {
+        legend.show <- FALSE
+    }
+
+    logo.urls <- getLogoUrls(logos, orig.x, scatter.labels, n)
+    labels.or.logos <- if (!is.null(logo.urls)) logo.urls else scatter.labels
+
+    footer <- processFooter(footer, scatter.labels.name, scatter.colors.name,
+                            scatter.sizes.name, scatter.mult.yvals, footer.wrap,
+                            footer.wrap.nchar)
+
+    # Convert axis to the appropriate type based on axis values and tick format
+    # Give warning where possible
+    x.axis.type <- getAxisType(x[not.na], x.tick.format)
+    x.tick.format <- checkD3Format(x.tick.format, x.axis.type, "X axis", convert = TRUE)
+    x <- convertAxis(x, x.axis.type)
+    y.axis.type <- getAxisType(y[not.na], y.tick.format)
+    y.tick.format <- checkD3Format(y.tick.format, y.axis.type, "Y axis", convert = TRUE)
+    y <- convertAxis(y, y.axis.type)
+    x.bounds.units.major <- getAxisBoundsUnitsMajor(x.tick.distance,
+                                                    x.tick.maxnum,
+                                                    x.bounds.maximum,
+                                                    x.bounds.minimum, x)
+    y.bounds.units.major <- getAxisBoundsUnitsMajor(y.tick.distance,
+                                                    y.tick.maxnum,
+                                                    y.bounds.maximum,
+                                                    y.bounds.minimum, y)
+
+    tooltips.text <- getTooltipsText(scatter.labels, not.na, x, y, x.tick.format,
+                                     x.tick.prefix, x.tick.suffix, y.tick.format,
+                                     y.tick.prefix, y.tick.suffix, scatter.sizes,
+                                     scatter.sizes.name, scatter.colors,
+                                     scatter.colors.name)
+
+    if (length(not.na) < n)
+        warning("Data points with missing values have been omitted.")
+
+    fit <- list()
+    if (fit.type != "None") {
+        x.axis.type <- getAxisType(unique(x[not.na]), x.tick.format)
+        fit <- fitLines(groups, x, y, not.na, fit.type, fit.ignore.last,
+                        fit.CI.show, fit.window.size, colors, fit.line.colors,
+                        fit.CI.colors, fit.CI.opacity, x.axis.type)
+    }
+
+    annotations <- processAnnotations(annotation.list, n, annot.data,
+                                      labels.or.logos)
+
+    Z <- if (is.null(scatter.sizes)) NULL else abs(scatter.sizes[not.na])
```

Comment:
Maybe call this `sizes` instead of `Z`? It seems to be the only uppercase variable here

Reply:
Renamed to `scatter.sizes` in 0fb193f3a766242fbf76605dd42d1d6eab748cd7.